src/components/input/input.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  forwardRef,
+  Component,
+  HostBinding,
+  Input,
+  Output,
+  Provider,
+} from 'angular2/core';
+import {EventEmitter} from 'angular2/src/facade/async';
+import {CONST_EXPR} from 'angular2/src/facade/lang';
+import {MdAriaTarget, MdAria} from '../../core/utilities/aria-directives';
+import {
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor
+} from 'angular2/src/common/forms/directives/control_value_accessor';
+import {BooleanFieldValue} from '../../core/annotations/field-value';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+const MD_INPUT_CONTROL_VALUE_ACCESSOR = CONST_EXPR(new Provider(
+    NG_VALUE_ACCESSOR, {
+      useExisting: forwardRef(() => MdInput),
+      multi: true
+    }));
+
+
+let uniqueId = 0;
+
+
+/**
+ * <md-input> component.
+ */
+@Component({
+  selector: 'md-input',
+  templateUrl: 'components/input/input.html',
+  styleUrls: ['components/input/input.css'],
+  providers: [MdAria, MD_INPUT_CONTROL_VALUE_ACCESSOR],
+  directives: [MdAriaTarget],
+})
+export class MdInput implements ControlValueAccessor {
+  private _empty: boolean = true;
+  private _focused: boolean = false;
+  private _onTouched: () => void = () => {};
+  private _onChangeFn: (_: any) => void = (_: any) => {};
```

Comment:
Okay so after looking deeper, it _is_ invalid Dart to assign a `function(){}` to one that expects an argument. `ts2dart` though emits `dynamic` typing in this case, which remove type checking. I've updated the code to use `noop`, but note this might become invalid in the future.